gateway/src/cluster/impl.rs

```diff
@@ -311,6 +311,12 @@ impl Cluster {
                 shard_config.sequence = Some(data.sequence);
             }
 
+            if let Some(shard_presence) = &config.shard_presence {
+                if let Some(shard_presence) = shard_presence(idx) {
+                    shard_config.presence = Some(shard_presence);
+                }
+            }
+
```

Comment:
The check whether the `shard_presence` closure returns `Some` is not necessary. 

```suggestion
            if let Some(shard_presence) = &config.shard_presence {
                shard_config.presence = shard_presence(idx);
            }

```

Reply:
it's not, if the method returns none it should default back to the main presence

Reply:
I personally would assume when I return `None` that it get set to nothing 🤔

Reply:
i'll document it